modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/EmptyBucketGetUserAssignmentTest.java

```diff
@@ -0,0 +1,99 @@
+import com.intuit.wasabi.tests.library.util.serialstrategies.DefaultNameExclusionStrategy;
+import com.intuit.wasabi.tests.library.util.serialstrategies.DefaultNameInclusionStrategy;
+import com.intuit.wasabi.tests.model.Assignment;
+import com.intuit.wasabi.tests.model.Bucket;
+import com.intuit.wasabi.tests.model.Experiment;
+import com.intuit.wasabi.tests.model.User;
+import com.intuit.wasabi.tests.model.factory.AssignmentFactory;
+import com.intuit.wasabi.tests.model.factory.BucketFactory;
+import com.intuit.wasabi.tests.model.factory.ExperimentFactory;
+import com.intuit.wasabi.tests.model.factory.UserFactory;
+
+/**
+ * A test to check if user can be assigned if the previous assignment bucket is empty
+ */
+public class EmptyBucketGetUserAssignmentTest extends TestBase {
+
+    private Experiment experiment;
+    private List<Bucket> buckets = new ArrayList<>();
+    private User specialUser = UserFactory.createUser("SpecialForBucketTest");
+    private User specialUser2 = UserFactory.createUser("Special2ForBucketTest");
+
+    /**
+     * Initializes a default experiment.
+     */
+    public EmptyBucketGetUserAssignmentTest() {
+        setResponseLogLengthLimit(1000);
+
+        experiment = ExperimentFactory.createExperiment();
+
+        DefaultNameExclusionStrategy experimentComparisonStrategy = new DefaultNameExclusionStrategy("creationTime", "modificationTime", "ruleJson");
+        experiment.setSerializationStrategy(experimentComparisonStrategy);
+
+    }
+
+    @Test(dependsOnGroups = {"ping"})
+    public void t_GetUserAssignmentEmptyBucket() {
+        Experiment exp = postExperiment(experiment);
+        Assert.assertNotNull(exp.creationTime, "Experiment creation failed (No creationTime).");
+        Assert.assertNotNull(exp.modificationTime, "Experiment creation failed (No modificationTime).");
+        Assert.assertNotNull(exp.state, "Experiment creation failed (No state).");
+        experiment.update(exp);
+        buckets = BucketFactory.createBuckets(experiment, 3);
+        postBuckets(buckets);
+        
+        experiment.state = Constants.EXPERIMENT_STATE_RUNNING;
+        Experiment exp2 = putExperiment(experiment);
+        assertEqualModelItems(exp2, experiment);
+        experiment.update(exp);
+        
+        // special user 2 assigned to bucket 0
+        Assignment assignment = AssignmentFactory.createAssignment()
+                .setAssignment(buckets.get(0).label)
+                .setExperimentLabel(experiment.label)
+                .setOverwrite(true);
+        Assignment putAssignmentFor2 = putAssignment(experiment, assignment, specialUser2);
+        assertEqualModelItems(putAssignmentFor2, assignment, new DefaultNameInclusionStrategy("assignment"));
+
+       // Empty bucke to which use is assigned
+       List<Bucket> emptyBucket = new ArrayList<>();
+       emptyBucket.add(buckets.get(0));
+       emptyBucket = putBucketsState(emptyBucket, Constants.BUCKET_STATE_EMPTY);
```

Comment:
You don't need to assign the return value of putBucketsState (and the comment has a little typo "bucke(t)").